src/components/Feedback.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from "react";
+import { Button, createTheme, ThemeProvider, styled, ButtonProps } from "@material-ui/core";
+import { Feedback as FeedbackIcon } from "@material-ui/icons";
+import { indigo500 } from "material-ui/styles/colors";
+import { FeedbackDialog } from "./FeedbackDialog";
+import { useBooleanState } from "../hooks/useBoolean";
+import { ButtonPosition, FeedbackOptions } from "../domain/entities/Feedback";
+import { SubmitDialog } from "./SubmitDialog";
+import i18n from "../locales";
+
+interface FeedbackProps {
+    options?: FeedbackOptions;
+    username?: string;
+}
+
+export const Feedback: React.FC<FeedbackProps> = React.memo(({ options, username }) => {
+    const [showDialog, { close: closeDialog, open: openDialog }] = useBooleanState(false);
+    const [showSDialog, { close: closeSDialog, open: openSDialog }] = useBooleanState(false);
+    const [contentSDialog, setContentSDialog] = useState("");
```

Comment:
I'd write `S` -> `Submit`, short names only for very small blocks where the context is very clear.